server/utils/logger.ts

```diff
@@ -166,8 +169,109 @@ if (config.get('logging.file.enabled')) {
     level: config.get('logging.file.level'),
     path: logPath,
   })
+}
+
+async function processStroomFiles() {
+  const stroomFolder = resolve(appRoot, config.get('logging.stroom.folder'))
+  const processingFolder = resolve(stroomFolder, 'processing')
+
+  const files = (
+    await fsPromise.readdir(stroomFolder, {
+      withFileTypes: true,
+    })
+  ).filter((item) => item.isFile())
+
+  for (const file of files) {
+    const from = resolve(stroomFolder, file.name)
+    const to = resolve(processingFolder, file.name)
+
+    const { size } = await fsPromise.stat(from)
+
+    if (size < 0) {
+      continue
+    }
+
+    await fsPromise.rename(from, to)
+  }
+
+  const processing = (
+    await fsPromise.readdir(processingFolder, {
+      withFileTypes: true,
+    })
+  ).filter((item) => item.isFile())
+
+  for (const file of processing) {
+    const name = resolve(processingFolder, file.name)
+    try {
+      await sendLogsToStroom(name)
+    } catch (e) {
+      // ironically we cannot use our logger here.
+      console.error(e, 'Unable to send logs to ACE')
+    }
+  }
+}
+
+async function sendLogsToStroom(file: string) {
+  const logBuffer = await fsPromise.readFile(file)
+
+  const res = await fetch(config.get('logging.stroom.url'), {
+    method: 'POST',
+    body: await gzip(logBuffer),
+    headers: {
+      Compression: 'gzip',
+      Feed: config.get('logging.stroom.feed'),
+      Environment: config.get('logging.stroom.environment'),
+      System: config.get('logging.stroom.system'),
+    },
+  })
+
+  if (!res.ok) {
+    throw new Error('Failed to send logs to stroom')
+  }
+
+  await fsPromise.unlink(file)
+}
+
+if (config.get('logging.stroom.enabled')) {
+  const stroomFolder = resolve(appRoot, config.get('logging.stroom.folder'))
+  const processingFolder = resolve(stroomFolder, 'processing')
+  const processedFolder = resolve(stroomFolder, 'processed')
+
+  let date = new Date()
+
+  if (!fs.existsSync(stroomFolder)) {
+    fs.mkdirSync(stroomFolder, { recursive: true })
+  }
+
+  if (!fs.existsSync(processingFolder)) {
+    fs.mkdirSync(processingFolder, { recursive: true })
+  }
 
-  console.log(logPath, dirname(logPath))
+  if (!fs.existsSync(processedFolder)) {
+    fs.mkdirSync(processedFolder, { recursive: true })
+  }
+
+  const stroomStream = new WritableStream({
+    async write(message) {
+      if (message.code) {
+        const path = resolve(stroomFolder, `logs.${+date}.txt`)
+        await fsPromise.appendFile(path, JSON.stringify(message) + '\n')
```

Comment:
Our eslint rules will want this to be a template string instead of using `+`
```
await fsPromise.appendFile(path, `${JSON.stringify(message)}\n`)
```